src/ripple/shamap/SHAMapNodeID.h

```diff
@@ -21,159 +21,128 @@
 #define RIPPLE_SHAMAP_SHAMAPNODEID_H_INCLUDED
 
 #include <ripple/basics/base_uint.h>
-#include <ripple/beast/utility/Journal.h>
-#include <ripple/protocol/Serializer.h>
+#include <optional>
 #include <ostream>
 #include <string>
 #include <tuple>
 
 namespace ripple {
 
-// Identifies a node in a half-SHA512 (256 bit) hash map
+/** Identifies a node inside a SHAMap */
 class SHAMapNodeID
 {
 private:
-    uint256 mNodeID;
-    int mDepth;
+    uint256 id_;
+    unsigned int depth_ = 0;
 
 public:
-    SHAMapNodeID();
-    SHAMapNodeID(int depth, uint256 const& hash);
-    SHAMapNodeID(void const* ptr, int len);
+    SHAMapNodeID() = default;
+    SHAMapNodeID(SHAMapNodeID const& other) = default;
+    SHAMapNodeID(unsigned int depth, uint256 const& hash);
 
-    bool
-    isValid() const;
-    bool
-    isRoot() const;
-
-    // Convert to/from wire format (256-bit nodeID, 1-byte depth)
-    void
-    addIDRaw(Serializer& s) const;
-    std::string
-    getRawString() const;
-
-    bool
-    operator==(const SHAMapNodeID& n) const;
-    bool
-    operator!=(const SHAMapNodeID& n) const;
+    SHAMapNodeID& operator=(SHAMapNodeID const& other) = default;
 
     bool
-    operator<(const SHAMapNodeID& n) const;
-    bool
-    operator>(const SHAMapNodeID& n) const;
-    bool
-    operator<=(const SHAMapNodeID& n) const;
-    bool
-    operator>=(const SHAMapNodeID& n) const;
+    isRoot() const
```

Comment:
Stylistic:  This is reverting a style that I find better, but I'm not going to loose sleep over it.

I think it is better to define member functions outside of the class declaration even when they are inline so that the reader can more easily get an overview of the class API.